operators/hack/run-e2e.sh

```diff
@@ -33,7 +33,22 @@ sed \
     kubectl apply -f -
 
 # retrieve pod responsible for running the job
-pod=$(kubectl get pods -n $NAMESPACE --selector=job-name=$JOB_NAME --output=jsonpath={.items..metadata.name})
+pod=""
+retry=0
+e2e_pod_creation_timeout=30
+while true; do
+    if [[ ${retry} -ge ${e2e_pod_creation_timeout} ]]; then
+        echo "failed to get the e2e pod name after ${e2e_pod_creation_timeout} seconds"
+        exit 1
+    fi
+    ((retry++))
+    pod=$(kubectl get pods -n ${NAMESPACE} --selector=job-name=${JOB_NAME} --output=jsonpath={.items..metadata.name})
+    if [[ ! -z "${pod}" ]]; then
+        break
+    fi
+    sleep 1;
+done
+
```

Comment:
Do you see a problem to rather use?
```bash
kubectl wait pods -n $NAMESPACE -l job-name=$JOB_NAME --for condition=Ready --timeout 30s
```

Reply:
I was thinking that `wait` does not work if the `pod` does not exist (which is the case here)
Can you confirm that it will wait if the pod is not yet created ?

Reply:
You are completely right. I forgot that.